src/mcts/search.cc

```diff
@@ -956,7 +956,11 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
         ++possible_moves;
       }
       const float Q = child.GetQ(fpu);
-      const float score = child.GetU(puct_mult) + Q;
+      const float tanhU = tanh(child.GetU(puct_mult));
+      float score = tanhU + Q;
+      if (tanhU * Q != -1) {
+        score = score / (tanhU * Q + 1);
```

Comment:
Division by zero is well-defined, in the sense that it leads to +inf or -inf, depending on the sign of the numerator, so no need for the check. Moreover, abs(tanhU) will never attain 1 for all practical values of U, which would be a necessary condition for the product to be -1.

Reply:
Thanks. I was thinking U*Q could be -1, but tanhU*Q shouldn't have that problem.